seqr/views/utils/file_utils.py

```diff
@@ -86,25 +91,32 @@ def save_uploaded_file(request, process_records=None):
         raise ValueError("Received %s files instead of 1" % len(request.FILES))
 
     # parse file
-    stream = request.FILES.values()[0]
+    stream = next(iter(request.FILES.values()))
     filename = stream._name
 
+    if filename.endswith('.csv'):
+        try:
+            a = unicode
+        except:
+            stream = TextIOWrapper(stream.file, encoding = 'utf-8')
+    elif not filename.endswith('.xls') and not filename.endswith('.xlsx'):
+        stream = TextIOWrapper(stream.file, encoding = 'utf-8')
```

Comment:
This a little ugly solution: if `unicode` is defined then do nothing for `.csv` file, otherwise wrap all non-excel files with `TextIOWrapper`.
We can't do the same thing as `file_iter()` data because uploaded data (`bytes`) can only be converted to `unicode` for Python 2. So we have to keep the data in `bytes` type otherwise they won't work with `csv.reader`.

Reply:
this should be done in `parse_file`, not here

Reply:
also, when you move it this is a really messy way to do it. what does `a = unicode` even do? `a` is never used and how would that assignment ever throw an exception?

Reply:
`a = unicode` is for testing if `unicode` is defined or not. Maybe a better way is to use `unicode = unicode` or just `unicode`.